key-script

```diff
@@ -22,16 +22,19 @@ message()
 
 check_yubikey_present="$(ykinfo -q -"$YUBIKEY_LUKS_SLOT")"
 
-if [ -z "$cryptkeyscript" ]; then
-    if [ -x /bin/plymouth ] && plymouth --ping; then
-        cryptkeyscript="plymouth ask-for-password --prompt"
-    else
-        cryptkeyscript="/lib/cryptsetup/askpass"
-    fi
+if [ -z "$YUBIKEY_CHALLENGE" ] || [ "$check_yubikey_present" -ne "1" ] ; then
```

Comment:
Just checking for the logic:

I will most probably never use 1FA.
So my ``YUBIKEY_CHALLENGE``  will always be empty and I will always enter the if-branch, no matter, whether I have plugged in the yubikey or not.

But if you want to use 1FA, your YUBIKEY_CHALLENGE will not be empty. So the first condition will be false, but in case the check_yubikey_preset is false (no yubikey plugged in) you want to be asked for the passphrase, right?

Please see line 38 and change the condition of the string comparison to

~~~~bash
    ... || [ "$check_yubikey_preset" != "1" ]; then
~~~~

Thanks.

Reply:
will be in next update to pull request. thanks for spotting this. has caused edge case issue recently on my side that i had not previously caught (prompt for password when key not present).

Reply:
and yes the interpretation of the logic that you described is the intent.